flake/nixvim.nix

```
@@ -0,0 +1,49 @@
+{
+  inputs,
+  lib,
+  self,
+  ...
+}:
+{
+  imports = [
+    inputs.nixvim.flakeModules.default
+  ];
+
+  nixvim =
+    let
+      nameFunction = lib.id;
+    in
+    {
+      packages = {
+        enable = true;
+        inherit nameFunction;
+      };
+      checks = {
+        enable = true;
+        inherit nameFunction;
+      };
+    };
+
+  flake.nixvimModules = {
+    default = ../modules/nixvim;
+  };
+
+  perSystem =
+    { system, ... }:
+    {
+      nixvimConfigurations = {
+        khanelivim = inputs.nixvim.lib.evalNixvim {
+          inherit system;
+
+          extraSpecialArgs = {
+            inherit inputs system;
+            inherit (inputs) self;
```

Why not inherit directly: 
```suggestion
            inherit self;
```

You have it available as a module arg and also use it below.

---

Was just a copy paste from the old code